### PR TITLE
Dgrams stats

### PR DESCRIPTION
**Pull request by ddragana**

### CODE REVIEW

**Comment by ddragana** [comment]
This is based on #1220 and only the last push is new.

File: neqo-transport/src/stats.rs [hunk 1]
```diff
@@ -127,6 +127,17 @@ pub struct Stats {
...
+    /// The number of incoming datagrams dropped due to reaching the limit
+    /// of the incoming queue.
+    pub incoming_datagram_dropped: usize,
+    /// The number of datagrams declared lost.
+    pub outgoing_datagram_lost: usize,
+    /// The number of datagrams dropped due to being too large.
+    pub outgoing_datagram_dropped_too_big: usize,
+    /// The number of datagrams dropped due to reaching the limit of the
+    /// outgoing queue.
+    pub outgoing_datagram_dropped_queue_full: usize,
```
Comment (martinthomson) [inline neqo-transport/src/stats.rs:140]: It seems like we have a datagram_tx struct needed here to capture the different outcomes.  We won't avoid a train wreck of names at the usage site, but we might be able to limit some of the names from getting too unwieldy.

File: neqo-transport/src/events.rs [hunk 2]
```diff
@@ -168,7 +168,7 @@ impl ConnectionEvents {
 
     // The number of datagrams in the events queue is limited to max_queued_datagrams.
     // This function ensure this and deletes the oldest datagrams if needed.
-    fn check_datagram_queued(&self, max_queued_datagrams: usize) {
+    fn check_datagram_queued(&self, max_queued_datagrams: usize, stats: &mut Stats) {
         let mut q = self.events.borrow_mut();
         let mut remove = None;
         if q.iter()
```
Comment (martinthomson) [inline neqo-transport/src/events.rs:174]: Before we ship this, we really need to ensure that we don't keep this bit.  We should just be able to do a simple comparison of `self.datagrams.len()` (or something similarly named) with a target value.

**Review by martinthomson — COMMENTED** [review]
This part looks good.  The naming is again hard, but the code is solid.

File: neqo-transport/src/events.rs (hunk 2 above)
Comment (ddragana) [reply neqo-transport/src/events.rs:174]: Yes, open #1218 to fix this. That is the next on my list to fix.

**Pull request merged**